ci-bench/src/bench_lib/compare.rs

```diff
@@ -0,0 +1,181 @@
+            missing.push(scenario.clone());
+            continue;
+        };
+
+        let diff = instr_count as i64 - baseline_instr_count as i64;
+        let diff_ratio = diff as f64 / baseline_instr_count as f64;
+        diffs.push(Diff {
+            scenario: scenario.clone(),
+            baseline: baseline_instr_count,
+            candidate: instr_count,
+            diff,
+            diff_ratio,
+        });
+    }
+
+    CompareResult {
+        diffs,
+        missing_in_baseline: missing,
+    }
+}
+
+/// Prints a report of the comparison to stdout, using GitHub-flavored markdown
+fn report(mut result: CompareResult) {
+    result.diffs.sort_by(|diff1, diff2| {
+        diff2
+            .diff_ratio
+            .abs()
+            .total_cmp(&diff1.diff_ratio.abs())
+    });
+
+    println!("# Benchmark results");
+
+    if !result.missing_in_baseline.is_empty() {
+        println!("### ⚠️ Warning: missing benchmarks");
+        println!();
+        println!("The following benchmark scenarios are present in the candidate but not in the baseline:");
+        println!();
+        for scenario in &result.missing_in_baseline {
+            println!("* {scenario}");
+        }
+    }
+
+    if result.diffs.is_empty() {
+        println!("### ⚠️ Warning: missing benchmarks");
+        println!();
+        println!("There are no benchmarks to report");
+        return;
+    }
+
+    let (noteworthy, negligible) = split_on_threshold(&result.diffs);
+
+    println!("### Noteworthy instruction count differences");
+    if noteworthy.is_empty() {
+        println!(
+            "_There are no noteworthy instruction count differences (i.e. above {}%)_",
+            CHANGE_THRESHOLD * 100.0
+        );
+    } else {
+        table(noteworthy);
+    }
+
+    println!("### Other instruction count differences");
+    if negligible.is_empty() {
+        println!("_There are no other instruction count differences_");
+    } else {
+        println!("<details>");
+        println!("<summary>Click to expand</summary>\n");
+        table(negligible);
+        println!("</details>\n")
+    }
+}
+
+/// Splits the diffs into two slices, the first one containing the diffs that exceed the threshold,
+/// the second one containing the rest
+///
+/// Assumes that the diff slice is sorted by `diff_ratio` in descending order.
+fn split_on_threshold(diffs: &[Diff]) -> (&[Diff], &[Diff]) {
+    match diffs
+        .iter()
+        .position(|diff| diff.diff_ratio < CHANGE_THRESHOLD)
+    {
+        None => (diffs, &[]),
+        Some(first_below_threshold) => (
+            &diffs[..first_below_threshold],
+            &diffs[first_below_threshold..],
+        ),
+    }
+}
+
+/// Renders the diffs as a markdown table
+fn table(diffs: &[Diff]) {
+    println!("| Scenario | Baseline | Candidate | Diff ");
+    println!("| --- | --- | --- | --- |");
+    for diff in diffs {
+        println!(
+            "| {} | {} | {} | {} ({:.2}%) |",
```

Comment:
Can we do right-aligned numbers in Markdown? What about table cell background colors (green for significantly faster, red for significantly slower)?

Reply:
Aligning is possible, but colors aren't. The Rust CI uses emoji to work around this. We could do something like follows. What do you think?

### Regression

| Scenario | Baseline | Candidate | Diff 
| --- | ---: | ---: | ---: |
| transfer_no_resume_1.2_rsa_aes_client | 7767624 | 7789541 | ⚠️ 21917 (0.28%) |
| transfer_no_resume_1.2_rsa_aes_server | 8775849 | 8798099 | ⚠️ 22250 (0.25%) |

### Improvement

| Scenario | Baseline | Candidate | Diff 
| --- | ---: | ---: | ---: |
| transfer_no_resume_1.2_rsa_aes_client | 7767624 | 7789541 | ✅ -21917 (-0.28%) |
| transfer_no_resume_1.2_rsa_aes_server | 8775849 | 8798099 | ✅ -22250 (-0.25%) |

Reply:
Emoji seems good!